src/BandedPlusSemiseparableMatrices.jl

```diff
@@ -0,0 +1,372 @@
+struct BandedPlusSemiseparableMatrix{T} <: LayoutMatrix{T}
+    bands::BandedMatrix{T}
+    lowerfill::LowRankMatrix{T} 
+    upperfill::LowRankMatrix{T}
+end
+
+BandedPlusSemiseparableMatrix(B, (U,V), (W,S)) = BandedPlusSemiseparableMatrix(B,  LowRankMatrix(U, V'), LowRankMatrix(W, S'))
+
+size(A::BandedPlusSemiseparableMatrix) = size(A.bands)
+copy(A::BandedPlusSemiseparableMatrix) = A # not mutable
+
+function getindex(A::BandedPlusSemiseparableMatrix, k::Integer, j::Integer)
+    if j > k 
+        A.upperfill[k,j] + A.bands[k,j]
+    elseif k > j
+        A.lowerfill[k,j] + A.bands[k,j]
+    else
+        A.bands[k,j]
+    end
+end
+
+
+"""
+Represents factors matrix for QR for banded+semiseparable. we have
+
+    for j = 0, we have A₀=tril(UV',-1) + B + triu(WS',1);
+    F = qrfactUnblocked!(A₀).factors # full QR factors;
+    As j increases,  for BandedPlusSemiseparableQRPerturbedFactors A:
+    A[1:j,:] == F[1:j,:];
+    A[:,1:j] == F[:,1:j];
+    A[k, k] == F[k, k] for k < j;
+    A[j+1:end,j+1:end] == A₀[j+1:end,j+1:end] + U[j+1:end,:]*Q*S[j+1:end]' + U[j+1:end,:]*K*U[j+1:end]'*A₀[j+1:end,j+1:end]
+                          + U[j+1:end,:]*[Eₛ 0] + [Xₛ;0]*S[j+1:end]' + [Yₛ;0]*U[j+1:end]'*A₀[j+1:end,j+1:end] + [Zₛ 0;0 0]
+"""
+
+struct BandedPlusSemiseparableQRPerturbedFactors{T} <: LayoutMatrix{T}
+    n::Int # matrix size
+    r::Int # lower rank
+    p::Int # upper rank
+    l::Int # lower bandwidth
+    m::Int # upper bandwidth
+    U::Matrix{T} # n × r
+    V::Matrix{T} # n × r
+    W::Matrix{T} # n × (p+r)
+    S::Matrix{T} # n × (p+r)
+    B::BandedMatrix{T} # lower bandwidth l and upper bandwidth l + m
+
+    Q::Matrix{T} # r × p
+    K::Matrix{T} # r × r
+    Eₛ::Matrix{T} # r × min(l+m,n)
+    Xₛ::Matrix{T} # min(l,n) × p
+    Yₛ::Matrix{T} # min(l,n) × r
+    Zₛ::Matrix{T} # min(l,n) × min(l+m,n)
+
+    j::Base.RefValue{Int} # how many columns have been upper-triangulised
+end
+
+BandedPlusSemiseparableMatrix(A::BandedPlusSemiseparableQRPerturbedFactors) = BandedPlusSemiseparableMatrix(A.B, (A.U,A.V), (A.W,A.S))
+BandedPlusSemiseparableQRPerturbedFactors(A::BandedPlusSemiseparableMatrix) = BandedPlusSemiseparableQRPerturbedFactors(copy(A.lowerfill.args[1]), copy(A.lowerfill.args[2]'), copy(A.upperfill.args[1]), copy(A.upperfill.args[2]'), copy(A.bands))
+
+size(A::BandedPlusSemiseparableQRPerturbedFactors) = (A.n,A.n)
+
+function BandedPlusSemiseparableQRPerturbedFactors(U,V,W,S,B)
```

Comment:
Make constructor consistent with `BandedPlusSemiseparableMatrix`:
```suggestion
function BandedPlusSemiseparableQRPerturbedFactors(B, (U,V), (W,S))
```